src/libp2p/rpc/decoder.rs

```diff
@@ -44,31 +44,32 @@ where
         // https://github.com/mxinden/asynchronous-codec/blob/master/src/framed_read.rs#L161
         let mut buf = [0u8; 8 * 1024];
         loop {
-            let n = std::task::ready!(Pin::new(&mut self.io).poll_read(cx, &mut buf))?;
+            let this = self.as_mut().project();
+            let n = std::task::ready!(this.io.poll_read(cx, &mut buf))?;
             // Terminated
             if n == 0 {
                 let item = serde_ipld_dagcbor::de::from_reader(&self.bytes[..])
                     .map_err(|e| io::Error::new(io::ErrorKind::Other, e.to_string()));
                 return Poll::Ready(item);
             }
-            self.bytes_read += n;
-            if self.max_bytes_allowed > 0 && self.bytes_read > self.max_bytes_allowed {
+            *this.bytes_read += n;
```

Comment:
Why do you use C++ in Rust?

Reply:
Haha, if only I could do `*this->bytes_read++;`